modules/internal/ChapelBase.chpl

```diff
@@ -983,6 +983,20 @@ module ChapelBase {
   // sync statements and for joining coforall and cobegin tasks
   //
 
+  inline proc chpl_rt_reset_task_spawn() {
+    pragma "fn synchronization free"
+    extern proc chpl_task_reset_spawn_order();
+    chpl_task_reset_spawn_order();
+  }
+
+  proc chpl_resetTaskSpawn(numTasks) {
+    const dptpl = if dataParTasksPerLocale==0 then here.maxTaskPar
+                  else dataParTasksPerLocale;
+
+    if numTasks >= dptpl then
+      chpl_rt_reset_task_spawn();
```

Comment:
Any reason not to call `chpl_task_reset_spawn_order()` directly here, instead of this intermediary?

Reply:
Not a great reason, but I've been doing other task placement experiments and it has been useful to be able to manually call this.